web/modules/custom/nys_comment/nys_comment.module

```diff
@@ -175,3 +175,36 @@ function nys_comment_deploy_default_rejected_value() {
     $comment->save();
   }
 }
+
+/**
+ * Implements hook_cron().
+ */
+function nys_comment_cron() {
+  $content_types = ['bill'];
+  $current_session_year = \Drupal::config('nys_config.settings')->get('nys_session_year');
+
+  // Get all nodes with open comments created outside current session.
+  $query = Drupal::database()->select('node', 'n');
+  $query->join('node__field_ol_session', 'ns', 'n.nid = ns.entity_id');
+  $query->join('node__field_comments', 'nc', 'n.nid = nc.entity_id');
+  $query->fields('n', ['nid', 'type']);
+  $query->fields('ns', ['field_ol_session_value']);
+  $query->condition('nc.field_comments_status', '1', '!=');
+  $query->condition('n.type', $content_types, 'IN');
+  $query->condition('ns.field_ol_session_value', $current_session_year, '<=');
```

Comment:
I think this needs to be `<` rather than `<=`.  If the current session is 2023, any bills from 2023/4 should still be allowed to receive comments.